### PR TITLE
Export DDSes in namespaced way in fluid-framework

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR is an experiment in namespacing (i don't know if that's the right technical term here) the contents of fluid-framework, especially the DDSes. This change is straightforward, but it does require downstream consumers to update their code to use the namespaces.

### CODE REVIEW

File: examples/hosts/app-integration/external-controller/src/app.ts [hunk 1]
```diff
@@ -11,12 +11,13 @@ import {
...
 import {
     IFluidContainer,
-    SharedMap,
+    map,
 } from "fluid-framework";
 import { v4 as uuid } from "uuid";
 import { DiceRollerController } from "./controller";
 import { makeAppView } from "./view";
 
+// const SharedMap = map.SharedMap;
```
Comment (tylerbutler) [inline examples/hosts/app-integration/external-controller/src/app.ts:20]:
```suggestion
```

**Comment by tylerbutler** [comment]
@ChumpChief This is an old draft, but I'd like to talk about this area for 1.0. It seems wrong that we export some but not all DDSes from the uber-package. I wanted to see what namespaces would look like, but I'm not a big fan of it as a solution. I would prefer to remove the re-exports and isolate DDSes in their own packages with possible exception of map/directory. What do you think?

**Comment by ChumpChief** [comment]
I am also not a fan of namespacing here.

I personally have no particularly strong opinions on DDSs being split into their own package(s) vs. included in the uber-package, but I think the current state was the product of some review and feedback cycles.  @skylerjokiel may have more insight on options weighed and feedback.

**Comment by tylerbutler** [comment]
Closing this PR since it was only for investigational purposes.

**Pull request closed without merging**